src/actions/sale/getSaleData.ts

```diff
@@ -0,0 +1,41 @@
+import { ethers } from "ethers";
+import { SalePhases } from "../../constants";
+import { Sale } from "../../contracts/types";
+import { SaleConfiguration, SaleData, SalePhase } from "../../types";
+
+export const getSaleData = async (
+  contract: Sale,
+): Promise<SaleData> => {
+  const salePhase: SalePhase =
+    (await contract.salePhase()) as SalePhase;
+  const salePhaseName = SalePhases[salePhase];
+  const saleStartTime = (await contract.saleStartBlockTimestamp()).toString();
+  const saleId = (await contract.saleId()).toNumber();
```

Comment:
`saleId` in `Sale` contract was calculated by keccak hash, so it will exceed the limitation of number in almost cases.
Use `string` or `BigNumber`